topic.go

```diff
@@ -349,6 +349,7 @@ func (t *Topic) validate(ctx context.Context, data []byte, opts ...PubOpt) (*Mes
 	}
 
 	msg := &Message{m, "", t.p.host.ID(), pub.validatorData, pub.local}
+	t.p.rt.PreValidation(t.p.host.ID(), []*Message{msg})
```

Comment:
While we're working on this area, how about we rename `PreValidation()` to something more accurate/obvious, e.g. `DispatchIDONTWANT()` or even something more generic like `Preprocess()` or `Intercept()`? The spirit of the method just isn't to validate (returns nothing), and it's confusing when juxtapose with `ValidateLocal()`.

Reply:
names, names, names 🤣

But yes, preprocessing is what it does.

Reply:
No strong opinion. `Preprocess` seems fine.